src/puppetlabs/trapperkeeper/internal.clj

```diff
@@ -562,6 +583,8 @@
         this)
       (a/start [this]
         (run-lifecycle-fns app-context s/start "start" ordered-services)
+        (when-let [restart-file (get-in-config (a/get-service this :ConfigService) [:global :restart-file])]
+          (inc-restart-counter! restart-file))
```

Comment:
I can't comment effectively on the line in question, but I think you'll *also* want to call this function below in the `a/restart` function.